grr/server/grr_response_server/gui/ui/components/client_details/client_details.ts

```diff
@@ -33,6 +102,116 @@ export class ClientDetails implements OnInit, OnDestroy {
     });
   }
 
+  getAccordionButtonState(
+      totalNumElements: number, currentMaxNumElementsShown: number,
+      initialMaxNumElementsShown: number): string {
+    if (totalNumElements > currentMaxNumElementsShown) {
+      return 'show-more';
+    } else if (totalNumElements <= initialMaxNumElementsShown) {
+      return 'no-button';
+    }
+    return 'show-less';
+  }
+
+  /**
+   * Returns the number of entries changed and a list of the aggregated changes
+   * descriptions (e.g. "2 users added")
+   */
+  static getDiffDescriptions(diff?: Diff<Client>[]): [number, string[]] {
+    if (diff === undefined) {
+      return [0, []];
+    }
+
+    let added: Map<string, number> = new Map();
+    let deleted: Map<string, number> = new Map();
+    let updated: Map<string, number> = new Map();
+    let diffDescriptions = [] as string[];
+    let numChanges = 0;
+
+    const classifyDiffItem = function(element: Diff<Client>) {
```

Comment:
use named function `function classifyDiffItem(...)` instead of assigning an unnamed function to a `const`. Also, do not use unnamed functions, but lambdas instead, because lambdas have cleaner scoping of `this`.